core/XH.js

```diff
@@ -332,14 +323,26 @@ class XHClass {
         }
     }
 
+    _idSeed = 1;
+
+    /**
+     * Generate an ID string, unique within this run of the client application and suitable
+     * for local-to-client uses such as auto-generated store record identifiers.
+     *
+     * Deliberately *not* intended to be globally unique, suitable for reuse, or to appear as such.
+     * @returns {string}
+     */
+    genId() {
+        return `xh-id-${this._idSeed++}`;
```

Comment:
I think we could use lodash's `uniqueId('xh-id-')` here and lose the local _idSeed